From 4aaa6795b07a38c408e2135fe9cd07dc71c1136e Mon Sep 17 00:00:00 2001
From: Sudarshan S <sudarshan.s@intel.com>
Date: Sun, 7 Feb 2021 18:21:32 +0530
Subject: [PATCH] vaDeriveImage: Enable WaDisableGmmLibOffsetInDeriveImage for
 JSL/EHL

Enables the software workaround to disable calculation of the UV
offset by gmmlib. An incorrect UV offset was by returned  gmmlib
in vaDeriveImage which casues the shift in the color planes(UV).
This was observed during a chromecast session in mirror mode on ChromeOS
with H264 encode.
---
 media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp b/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp
index e336cb377..bf231d6ba 100644
--- a/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp
+++ b/media_driver/linux/gen11/ddi/media_sku_wa_g11.cpp
@@ -413,6 +413,9 @@ static bool InitEhlMediaWa(struct GfxDeviceInfo *devInfo,
     MEDIA_WR_WA(waTable, Wa16KInputHeightNV12Planar420, 1);
     MEDIA_WR_WA(waTable, WaDisableCodecMmc, 1);
 
+    /*Software workaround to disable the UV offset calculation by gmmlib
+      CPU blt call will add/remove padding on the platform*/
+    MEDIA_WR_WA(waTable, WaDisableGmmLibOffsetInDeriveImage, 1);
     return true;
 }
 
